nexus/src/app/background/tasks/instance_watcher.rs

```diff
@@ -415,20 +445,22 @@ impl BackgroundTask for InstanceWatcher {
                 // handle the case where it's `None`, and I thought this was a
                 // bit neater...
                 let mut batch = batch.into_iter();
-                if let Some((mut curr_sled, instance, vmm, project)) = batch.next() {
-                    let mut client = mk_client(&curr_sled);
-                    let target = VirtualMachine::new(self.id, &curr_sled, &instance, &vmm, &project);
-                    tasks.spawn(self.check_instance(opctx, &client, target, vmm));
+                let mut curr_sled_id;
+                if let Some((sled, instance, vmm, project)) = batch.next() {
```

Comment:
Now that we're passing the sled ID into `check_instance` anyway, would it make sense just to have that function create its own client for each sled ID that gets passed to it? (This would also get rid of a `clone` in the callee.) I think that might also let us flatten this into a single for loop instead of having to call `next` up front first.

Reply:
So, each `reqwest` client (and therefore, each `progenitor`-generated client) owns a connection pool, allowing connection reuse. My intention behind cloning the client here was specifically to reuse pooled TCP connections to that sled-agent as previous checks finish, which won't happen if each `check_instance` task creates its own client. The `clone` here is lightweight as the `Client` is, internally, a reference-counted handle to that pool (as [described in the `reqwest` docs](https://docs.rs/reqwest/latest/reqwest/struct.Client.html)). 

I'm happy to add some commentary explaining that here --- there probably should've been some before!

Reply:
Oh, right, I forgot about how expensive client creation is. That makes sense. Yeah, I think a comment to that effect would be helpful here, but otherwise no need to change the loop structure. Thanks for the explanation!

Reply:
I'll definitely add a comment.

Reply:
Oh, you know what, there's actually been a comment here all along...which is what I had thought, I definitely remembered writing it: https://github.com/oxidecomputer/omicron/blob/3d8547071f8c80e29a7ec22bd4f8454507f4a38b/nexus/src/app/background/tasks/instance_watcher.rs#L439-L446

I think you just didn't see it while reviewing this because it wasn't part of the diff. Thanks, GitHub diff viewer...